packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -669,3 +670,44 @@ final GradleHandledError sslExceptionHandler = GradleHandledError(
   },
   eventLabel: 'ssl-exception-tag-mismatch',
 );
+
+final RegExp _unsupportedClassFileMajorVersionPattern = RegExp(r'Unsupported class file major version ([0-9]+)');
```

Comment:
For documentation can you add expected input here? 
Also I think you can match on  exactly 2 numbers https://javaalmanac.io/bytecode/versions/
That said if you want to match on numbers maybe use a \d+.